src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -35,8 +35,12 @@ class DeliverySessionService(
   val appointmentRepository: AppointmentRepository,
 ) {
   fun createUnscheduledSessionsForActionPlan(approvedActionPlan: ActionPlan) {
+    val previouslyApprovedSessions = approvedActionPlan.referral.approvedActionPlan?. let {
```

Comment:
Would it make sense to create a `approvedActionPlan.previousApprovedVersion` function? Then, in a separate PR, we could make that race-condition-resistant if we literally could walk the version history backwards